src/bp/core/events/middleware-chain.ts

```diff
@@ -13,20 +13,21 @@ const defaultOptions = {
 }
 
 export class MiddlewareChain {
-  private stack: { mw: sdk.IO.MiddlewareHandler; name: string }[] = []
+  private stack: { mw: sdk.IO.MiddlewareHandler; name: string; timeout: number }[] = []
 
   constructor(private options: MiddlewareChainOptions = defaultOptions) {
     this.options = { ...defaultOptions, ...options }
   }
 
-  use({ handler, name }: sdk.IO.MiddlewareDefinition) {
-    this.stack.push({ mw: handler, name })
+  use({ handler, name, timeout }: sdk.IO.MiddlewareDefinition) {
+    const timeoutInMs = timeout ? ms(timeout) : 0
+    this.stack.push({ mw: handler, name, timeout: timeoutInMs })
```

Comment:
`const timeoutInMs = timeout ? ms(timeout) : this.options.timeoutInMs` (and you can remove the || below)

Reply:
Good idea thanks!